src/Workspaces/Core/Portable/Serialization/SolutionReplicationContext.cs

```diff
@@ -4,18 +4,18 @@
 
 using System;
 using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Serialization
 {
     internal readonly struct SolutionReplicationContext : IDisposable
     {
-        private readonly ArrayBuilder<IDisposable> _resources;
+        private static readonly ObjectPool<ConcurrentSet<IDisposable>> s_pool = new(() => new());
 
-        private SolutionReplicationContext(ArrayBuilder<IDisposable> resources)
-            => _resources = resources;
+        private readonly ConcurrentSet<IDisposable> _resources;
```

Comment:
i was basically not at all confident that the old arraybuilder was not going to be used in a concurrent fashion.  So i moved to a concurrentset to be ultra paranoid about this being safe.